### PR TITLE
Implementing Algolia InstantSearch

### PR DESCRIPTION
**Pull request by Kosai106**
### Intro:
We've heard your cries for discoverability, and are proud to make this PR for you!

Not only did we implement search, but we even added voice search _(Chrome only)_ as a little bonus feature because we ❤️ Syntax.fm!

From all of us at [Algolia](https://algolia.com/), to all of you. Merry Christmas and happy holidays.

_And thanks for the numerous mentions on the podcast!_

### Features:
- InstantSearch
- Voice search
- Query highlighting
- Automatic indexing
- Mobile compatible

### Related:
- https://github.com/wesbos/Syntax/issues/120
- https://github.com/wesbos/Syntax/pull/168

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ const path = require('path');
 const glob = require('glob');
 
 module.exports = {
+  publicRuntimeConfig: { // Will be available on both server and client
+    ALGOLIA_APP_ID: '0BJH6ZHANS',
+    ALGOLIA_API_KEY: '6b603f518f8226fe184df623bba4dce1',
```
Comment (lannonbr) [inline next.config.js]: Is it safe to have this API key in the open?

Comment (Kosai106) [reply next.config.js]: Yes, there's no way around this - But you can't do anything with these except query data anyway.

**Comment by wesbos** [comment]
thank you so much! 

Can I ask if it's possible to show the show notes that match in the search somehow? Sort of like Google Search? If I search for React hooks I want to see where in the show notes they are mentioned

**Comment by Kosai106** [comment]
@wesbos That’s a good idea, we might be able to come up with a solution which should allow for this. It makes sense too considering your current search results also take the show notes into consideration as well.

**Comment by Kosai106** [comment]
Completely forgot to post here: After a lot of internal discussion, we haven't been able to come up with a way of dealing with highlighting inside the show notes currently.

I've already told @wesbos directly in person but figured I'd keep this PR updated with the information as well.

**Comment by itwasmattgregg** [comment]
I feel like the show note highlighting should hold up this PR. Would be nice to get this merged.

I'm assuming all the show notes are indexed? I see the show html is being saved but wonder if the show should be broken up more into separate sections for indexing. Also interesting that this is indexed client side. What was the reason behind that?

**Comment by stolinski** [comment]
We ended up going with flexsearch for the v2 of the site. Thank you

**Pull request closed without merging**